blog/2024-11-mastering-effect-and-afterRenderEffect/README.md

```diff
@@ -0,0 +1,477 @@
+      this.isEditMode.set(true);
+    } else {
+      this.bookForm.controls.isbn.enable();
+      this.isEditMode.set(false);
+    }
+  });
+
+  submitForm() {
+    // ...handle form submission logic
+  }
+}
+```
+
+In this example, `effect()` is ideal for handling the side effect (patching the form) without rerunning unnecessary computations. 
+We are also free to set signals in the effect now.
+To show that this is now perfectly valid, we updated another signal during that phase.
+We dediced for a signal called `isEditMode`, that is updated accordingly.
+In the past, we would have been using `ngOnChanges` to patch the form when the input was changed.
+
+
+### When to choose `effect()` over `computed()`
+
+The previous constraints on `effect()` have been removed, it is now more challenging to decide when to use `computed()` or `effect()`.
+In our opinion, the choice between `effect()` and `computed()` depends on the use case:
+- **Use `computed()`** for deriving a value based on other signals, especially when you need a pure, read-only reactive value. 
+  We covered `computed()` and `linkedSignal()` in this article: **[Angular 19: Introducing LinkedSignal for Responsive Local State Management](https://angular.schule/blog/2024-11-effect-and-afterRenderEffect)**
+- **Use `effect()`** if the operation is more complex, involves setting multiple signals or requires side effects to be performed outside the world of signals, such as synchronising reactive form states or logging events.
+
+For patching forms, there is currently no better approach than using effects. 
+This approach can also be easily migrated to what would have been done in the past with `ngOnchanges` – which is great.
+But whether we should have used a better computed signal for the `isEditMode` is questionable.
+It is not easy to make a decision here, because we could also have written the following:
+
+```ts
+isEditMode = computed(() => !!this.currentBook());
+```
+
+Perhaps we have to accept that in some situations both options are absolutely valid!
+
+
+## Introducing `afterRenderEffect()`
+
+The new `afterRenderEffect()` function allows us to control when tasks occur during the DOM update process.
+This is particularly beneficial for UI manipulations that require specific timing to avoid layout shifts and ensure smooth animations.
+
+The API itself mirrors the functionality of 
+* [`afterRender`](https://next.angular.dev/api/core/afterRender) *(registers a callback to be invoked each time the application finishes rendering)* and 
+* [`afterNextRender`](https://next.angular.dev/api/core/afterNextRender) *(Registers a callbacks to be invoked the next time the application finishes rendering, during the specified phases.)* 
+
+which are both in **Developer Preview**!
+
+The Angular docs recommend avoiding `afterRender` when possible and suggest specifying explicit phases with `afterNextRender` to avoid significant performance degradation. 
```

Comment:
```suggestion
The Angular docs recommend to avoid `afterRender` when possible and suggest to specify explicit phases with `afterNextRender` to avoid significant performance degradation. 
```